### PR TITLE
feat(@clayui/css): LPD-40160 Adds menubar-primary for CMS Product Menu variant

### PR DESCRIPTION
**Pull request by pat270**
https://liferay.atlassian.net/browse/LPD-40160

vertical-bar-primary

I'm sending this as a draft because we have a problem with the accordion headers. They can only be `button` and in some of the headers we have nested plus buttons. Having a `button` nested in a `button` violates:

> [Phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2), but there must be no [interactive content](https://html.spec.whatwg.org/multipage/dom.html#interactive-content-2) descendant and no descendant with the [tabindex](https://html.spec.whatwg.org/multipage/interaction.html#attr-tabindex) attribute specified.

from the w3c spec. I need some help figuring out which component we should place `<div role="button">` to make our markup valid.

### CODE REVIEW

**Comment by ethib137** [comment]
@marcoscv-work any recommendations for this one?

**Comment by marcoscv-work** [comment]
Hey guys @ethib137 @pat270 if we use `role="menuitem"` it should work fine.
An example here: https://codepen.io/marcoscv/full/NWQbJbx
You can change the <a> for a button and validate with Axe or Wave. it should be valid.

File: packages/clay-core/src/vertical-nav/VerticalNav.tsx [hunk 1]
```diff
@@ -242,53 +248,86 @@ function VerticalNav<T extends Record<string, any> | string>({
...
+									expandedKeys,
+									firstKey: collection.getFirstItem().key,
+									open,
+									spritemap,
+									toggle,
+								}}
+							>
+								<Collection<T> collection={collection} />
+							</VerticalNavContext.Provider>
+						</Nav>
```
Comment (matuzalemsteles) [inline packages/clay-core/src/vertical-nav/VerticalNav.tsx:300]: You could store this in a constant to avoid repeating the code and reuse it below.

Comment (pat270) [reply packages/clay-core/src/vertical-nav/VerticalNav.tsx]: Thanks @matuzalemsteles . I tried my best to put it in a `const`. I'm having trouble figuring out typescript after moving it.

File: packages/clay-core/src/vertical-nav/VerticalNav.tsx [hunk 2]
```diff
@@ -34,6 +34,16 @@ const Trigger = ({
 	</Button>
 );
 
+const Menubar = ({collection, value}) => {
```
Comment (pat270) [inline packages/clay-core/src/vertical-nav/VerticalNav.tsx:37]: @matuzalemsteles `yarn lint:typescript` is complaining here. 

```
packages/clay-core/src/vertical-nav/VerticalNav.tsx:37:19 - error TS7031: Binding element 'collection' implicitly has an 'any' type.

37 const Menubar = ({collection, value}) => {
                     ~~~~~~~~~~

packages/clay-core/src/vertical-nav/VerticalNav.tsx:37:31 - error TS7031: Binding element 'value' implicitly has an 'any' type.

37 const Menubar = ({collection, value}) => {
```
I don't know how to fix it.

File: packages/clay-core/src/nav/Link.tsx [hunk 3]
```diff
@@ -52,6 +52,40 @@ export const Link = React.forwardRef<
 		},
 		ref
 	) => {
+		if (showIcon) {
```
Comment (pat270) [inline packages/clay-core/src/nav/Link.tsx:55]: This change here will probably break tests in DXP due to `button` being changed to `div`.

File: packages/clay-core/src/vertical-nav/VerticalNav.tsx (hunk 2 above)
Comment (matuzalemsteles) [reply packages/clay-core/src/vertical-nav/VerticalNav.tsx:37]: I'll fix this, push a commit to deal with this and the const.

File: packages/clay-core/src/vertical-nav/VerticalNav.tsx (hunk 1 above)
Comment (matuzalemsteles) [reply packages/clay-core/src/vertical-nav/VerticalNav.tsx]: Hmm let me see, I'll push a commit dealing with that.

**Comment by ethib137** [comment]
Hey @pat270, what's the status on this one?

Also, it looks like keyboard navigation for vertical nav is broken by this PR.

**Comment by pat270** [comment]
@ethib137 I wasn't able to fix the the keyboard navigation. I had to rearrange the markup due to button in button React error. I believe the keyboard navigation isn't working for this because the text is nested inside other elements.

menubar-primary

**Comment by matuzalemsteles** [comment]
I might take a look at it this week if I can't get it moving @pat270.

**Comment by ethib137** [comment]
@matuzalemsteles and @pat270 we need this ASAP. Please prioritize getting this fixed and released. Thanks guys.

**Comment by pat270** [comment]
@ethib137 keyboard navigation works for the other vertical navs. It's only broken for the primary variant.

**Comment by ethib137** [comment]
That's odd... we need it to work for primary too.

**Comment by matuzalemsteles** [comment]
I fixed the navigation bug for primary, it wasn't working because we have a hook that handles navigation for our components but it was only being applied to the old VerticalNav and not to the new variant.

**Comment by pat270** [comment]
Thanks @matuzalemsteles! I'll update the tests

**Pull request merged**